SPEC.md

```diff
@@ -5373,18 +5452,108 @@ The following fully-qualified names exist when calling `workflow main` in `main.
 
-\* Task inputs are accessible to be set by the caller of `main` if the workflow is called with [`allowNestedInputs: true`](#computing-call-inputs) in its `meta` section.
+\* Task inputs are accessible to be set by the caller of `main` if the workflow is called with [`allow_nested_inputs: true`](#computing-call-inputs) in its `meta` section.
 
-### Call Statement
+### Workflow Inputs
+
+The workflow and [task `input` sections](#task-inputs) have identical semantics.
+
+### Workflow Outputs
+
+The workflow and [task `output` sections](#task-outputs) have identical semantics.
+
+By default, if the `output {...}` section is omitted from a top-level workflow, then the workflow has no outputs. However, the execution engine may choose allow the user to specify that when the top-level output section is omitted, all outputs from all calls (including nested calls) should be returned.
+
+If the `output {...}` section is omitted from a workflow that is called as a subworkflow, then that call must not have outputs. Formally defined outputs of subworkflows are required for the following reasons:
+
+- To present the same interface when calling subworkflows as when calling tasks.
+- To make it easy for callers of subworkflows to find out exactly what outputs the call is creating.
+- In the case of nested subworkflows, to give the outputs at the top level a simple fixed name rather than a long qualified name like `a.b.c.d.out` (which is liable to change if the underlying implementation of `c` changes, for example).
+
+### Workflow Hints
+
+The `hints` section is optional and may contain any number of attributes (key/value pairs) that provide hints to the execution engine. Some workflow hint keys are reserved and have well-defined values.
+
+The execution engine may ignore any unsupported hint. A workflow execution never fails due to the inability of the execution engine to recognize or satisfy a hint.
+
+#### Reserved Workflow Hints
+
+The following hints are reserved. An implementation is not required to support these attributes, but if it does support a reserved attribute it must enforce the semantics and allowed values defined below. The purpose of reserving these hints is to encourage interoperability of tasks and workflows between different execution engines.
+
+##### `allow_nested_inputs`
+
+* Allowed type: `Boolean`
+* Alias: `allowNestedInputs`
+
+By default, when a workflow calls a subworkflow or task it must provide inputs for all of the subworkflow/task inputs. However, if the runtime engine supports the `allow_nested_inputs` hint and it is set to `true` in a workflow's `hints` section, then that workflow is allowed to leave any subworkflow/task inputs unsatisfied. Any unsatisfied inputs must be specified by at runtime. If a runtime engine does not support `allow_nested_inputs` or if any inputs remain unsatisfied at runtime then the workflow fails with an error.
```

Comment:
> If a runtime engine does not support `allow_nested_inputs` or if any inputs remain unsatisfied at runtime then the workflow fails with an error

Isn't this counter to the statement above "A workflow execution never fails due to the inability of the execution engine to recognize or satisfy a hint"

Reply:
yes :)

Maybe we need a workflow requirements section as well/instead?

Reply:
As per discussion in the governance meeting today, this hint now only allows setting at runtime the values of optional inputs that are not explicitly specified in the call. Thus, there is never the case that a task/subworkflow input is unsatisfied, and so the workflow can execute regardless of the value of `allowed_nested_inputs`. I still provided the option for the workflow engine to refuse to execute a workflow if `allowed_nested_inputs` is `false` and the user tries to set a nested value anyway, but I consider that to be different than failing.